apis/cassandra/v1beta1/cassandradatacenter_webhook.go

```diff
@@ -150,9 +154,16 @@ func ValidateDatacenterFieldChanges(oldDc CassandraDatacenter, newDc CassandraDa
 		return attemptedTo("change serviceAccount")
 	}
 
-	// StorageConfig changes are disallowed
-	if !reflect.DeepEqual(oldDc.Spec.StorageConfig, newDc.Spec.StorageConfig) {
-		return attemptedTo("change storageConfig")
+	// CassandraDataVolumeClaimSpec changes are disallowed
+
+	if !reflect.DeepEqual(oldDc.Spec.StorageConfig.CassandraDataVolumeClaimSpec, newDc.Spec.StorageConfig.CassandraDataVolumeClaimSpec) {
```

Comment:
There is a semantic change here, not sure if this is OK. I can now change `MountPath` or `Name`whereas this was previously forbidden. I can also change `VolumeSource`, again not sure if this is intended.

Reply:
Well, I did think about this, but opted not to prevent it. When AdditionalVolumes are changed, the StatefulSets are recreated and these are allowed values to be changed in the StS (unlike PVC modifications).

Reply:
agreed, we should allow to remove such volumes 👍